patches/tModLoader/Terraria/ModLoader/UI/UIModSources.cs

```diff
@@ -266,14 +268,56 @@ private void ShowWelcomeMessage(string altButtonTextKey, string url, int gotoMen
 		});
 	}
 
-	private bool CheckDotnet()
+	string GetCommandToFindPathOfExecutable()
+	{
+		;
+		if (RuntimeInformation.IsOSPlatform(OSPlatform.Windows))
+			return "where";
+
+		if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux) ||
+		   RuntimeInformation.IsOSPlatform(OSPlatform.OSX) ||
+		   RuntimeInformation.IsOSPlatform(OSPlatform.FreeBSD))
+			return "which";
+
+		Logging.tML.Debug("Getting command for finding path of the executable failed due to an unsupported operating system");
+
+		return null;
+	}
+
+	string GetSystemDotnetPath()
+	{
+		string commandToFindPathOfExecutable = GetCommandToFindPathOfExecutable();
+		if (commandToFindPathOfExecutable == null)
+			return null;
+
+		try {
+			return Process.Start(new ProcessStartInfo {
+				FileName = commandToFindPathOfExecutable,
+				Arguments = "dotnet",
+				UseShellExecute = false,
+				RedirectStandardOutput = true
+			}).StandardOutput.ReadToEnd().Trim();
+		}
+		catch (Exception e) {
+			Logging.tML.Debug("Finding SystemDotnetPath failed: ", e);
+		}
+
+		return null;
+	}
+
+	bool IsCompatibleDotnetSdkAviable()
 	{
 		if (dotnetSDKFound)
 			return true;
 
+		string dotnetPath = GetSystemDotnetPath();
+		if (dotnetPath == null)
+			return false;
```

Comment:
Just change this to `GetSystemDotnetPath() ?? "dotnet"` so at least we get the old behavior as fallback

Reply:
https://github.com/tModLoader/tModLoader/pull/3334/commits/478eeee7c8d34c016806e230afe17b446c54b481